File: 503/next-greater-element-ii.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var nextGreaterElements = function (nums) {
    let len = nums.length;
    let ans = [];
    for (let i = 0; i < len; i++) {
        let j = i + 1;
        let found = false;
        while (j != i) {
            if (nums[j] > nums[i]) {
                found = true;
                ans.push(nums[j]);
                break;
            }
            j = (j + 1) % len;
        }
        if (!found) {
            ans.push(-1);
        }
    }
    return ans;
};

File: 503/next-greater-element-ii.py
class Solution:
    def nextGreaterElements(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        ans = []
        length = len(nums)
        for i, num in enumerate(nums):
            found = False
            j = (i+1) % length
            while j != i:
                if (nums[j] > num):
                    found = True
                    ans.append(nums[j])
                    break
                j = (j + 1) % length
            if not found:
                ans.append(-1)

        return ans
